src/middleware/routeHandler.ts

```diff
@@ -0,0 +1,250 @@
+import { backOff } from "exponential-backoff"
```

Comment:
regarding this:

> there needs to be a rollback handler to cater for the case of network failures that exist for a commit to 1 branch. else, we might have a scary situation whereby staging-lite updates, but staging doesnt, and this leads to a wrongly output production site

Correct me if otherwise, but we alr have a rollback mechanism right? the focus of thePR is to implement retry on in case rollback fails right?

Reply:
> but we alr have a rollback mechanism right

this is the rollback mechanism, but it wasnt working as intended for some call sites since never throw never throws the error -> leading to the the rollback not occuring as intended.